get_dir.c

```diff
@@ -0,0 +1,38 @@
+#include "shell.h"
+#include <dirent.h>
+
+char **get_dir(char *path)
+{
+    struct dirent *entry;
+    DIR *dir_pointer;
+    int counter;
+    char **dir_list;
+    if (!path)
+    {
+        printf("Error path");
+        exit(0);
+    }
+    counter = 0;
+    dir_list = malloc(1024);
+    if (dir_list == NULL)
+    {
+        exit(0);
+    }
+    dir_pointer = opendir(path);
+
+    if (dir_pointer == NULL)
+    {
+        free(dir_list);
+        perror("Error opening directory");
+        exit(0);
+    }
+
+    while ((entry = readdir(dir_pointer)) != NULL)
```

Comment:
Why is this the value of `entry` on the first iteration?
```
>>> info locals
entry = 0x556c23e6efe0
dir_pointer = 0x556c23e6efb0
counter = 0
dir_list = 0x556c23e66b60
>>> p *entry
$22 = {
  d_ino = 1333569,
  d_off = 1,
  d_reclen = 24,
  d_type = 4 '\004',
  d_name = ".\000\000\000\000|\177\027\017\000\000\000\000\002\000\000\000\000\000\000\000\030\000\004..\000\000\000BY\024\000\000\000\000\000\003\000\000\000\000\000\000\000\030\000\b[\000\000\000\000DY\024\000\000\000\000\000\004\000\000\000\000\000\000\000 \000\baddpart\000\000\000\000\000\000HY\024\000\000\000\000\000\005\000\000\000\000\000\000\000\030\000\bapt\000\000JY\024\000\000\000\000\000\006\000\000\000\000\000\000\000 \000\bapt-cache\000\000\000\000KY\024\000\000\000\000\000\a\000\000\000\000\000\000\000 \000\bapt-cdrom\000\000\000\000LY\024\000\000\000\000\000\b\000\000\000\000\000\000\000 \000\bapt-conf"...
}
```

Reply:
`d_name` looks like garbage.

Reply:
```
>>> p sizeof(entry->d_name)
$24 = 256
>>> ptype entry->d_name
type = char [256]

```

I guess it isn't garbage.  It just has more memory allocated than needed. As a `char [256]` data type, it is a null-terminated string: so the value of `entry->d_name` is `.` because `\000` terminates the string.